cmd/zapi/cmd/post/post.go

```diff
@@ -55,6 +55,9 @@ func (c *PostCommand) Run(args []string) (err error) {
 	if err != nil {
 		return err
 	}
+	if c.postFlags.shaperAST != nil {
+		c.logwriter.SetShaper(c.postFlags.shaperAST)
+	}
```

Comment:
Perhaps in this case create a `LogPostOpts` with the ast and pass this into `LogPostWriter`?

Reply:
Absolutely! Not sure why I didn't do that in the first place.

Reply:
done in 085394f24905391eff81c5d9a26e0017284d9352